.github/workflows/nugetUpload.yml

```diff
@@ -0,0 +1,30 @@
+name: 
+on:
+  push:
+    branches:
+      - master
+
+env:
+  NUGET_URL: https://dist.nuget.org/win-x86-commandline/v5.4.0/nuget.exe
+jobs:
+  windows:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Download Nuget.exe
+      run: Invoke-WebRequest -OutFile Nuget.exe $Env:NUGET_URL
+      
+    - name: Install Dependencies
+      run: ./Nuget.exe restore src/TgSharp.sln
+    
+    - name: setup-msbuild
+      uses: microsoft/setup-msbuild@v1
+
+    - name: Build & Upload Package
+      run: |
+        $date = get-date -format "yyyyMMdd-HHmm"
+        $hash = & git rev-parse --short HEAD
+        
+        nuget pack src/TgSharp.Core/TgSharp.Core.nuspec -Version "0.0.1-date.$date.git.$hash"
+        nuget push "TgSharp.0.0.1-date.$date.git.$hash.nupkg" {{secrets.NUGET_API_KEY}} -Source https://api.nuget.org/v3/index.json
```

Comment:
nitpick, as you used Nuget.exe above, let's use the same here too instead of just `nuget`